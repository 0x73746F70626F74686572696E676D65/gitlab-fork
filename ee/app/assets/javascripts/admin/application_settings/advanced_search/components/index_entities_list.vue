<script>
import { GlButton } from '@gitlab/ui';

export default {
  components: {
    GlButton,
  },
  props: {
    entities: {
      type: Array,
      required: false,
      default: () => [],
    },
    emptyText: {
      type: String,
      required: true,
    },
  },
};
</script>

<template>
  <div class="gl-border gl-rounded-small gl-overflow-auto h-12em gl-bg-white">
    <ul v-if="entities.length" class="gl-list-style-none gl-p-0 gl-my-2">
      <li v-for="entity in entities" :key="entity.id" class="gl-px-5 gl-py-2">
        <div class="gl-display-flex gl-align-items-center">
          <div class="gl-flex-grow-1 gl-pr-3 gl-truncate-end">{{ entity.text }}</div>
          <gl-button
            variant="danger"
            category="secondary"
            icon="remove"
            @click="$emit('remove', entity.id)"
          />
        </div>
      </li>
    </ul>
    <div v-else class="gl-h-full gl-display-flex gl--flex-center">
      {{ emptyText }}
    </div>
  </div>
</template>
