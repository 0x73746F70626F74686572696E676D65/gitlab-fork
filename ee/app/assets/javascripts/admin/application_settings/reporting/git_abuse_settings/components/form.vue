<script>
import { GlForm, GlFormGroup, GlFormInput, GlButton } from '@gitlab/ui';
import Api from 'ee/api';
import createFlash from '~/flash';

import {
  validateNumberOfRepos,
  validateReportingTimePeriod,
  validateExcludedUsers,
} from '../validations';

import {
  NUM_REPO_LABEL,
  NUM_REPO_DESCRIPTION,
  REPORTING_TIME_PERIOD_LABEL,
  EXCLUDED_USERS_LABEL,
  EXCLUDED_USERS_DESCRIPTION,
  SAVE_CHANGES,
  SUCCESS_MESSAGE,
  SAVE_ERROR_MESSAGE,
} from '../constants';

import UsersAllowlist from './users_allowlist.vue';

export default {
  name: 'GitAbuseRateLimitSettingsForm',
  components: {
    GlForm,
    GlFormGroup,
    GlFormInput,
    GlButton,
    UsersAllowlist,
  },
  i18n: {
    NUM_REPO_LABEL,
    NUM_REPO_DESCRIPTION,
    REPORTING_TIME_PERIOD_LABEL,
    EXCLUDED_USERS_LABEL,
    EXCLUDED_USERS_DESCRIPTION,
    SAVE_CHANGES,
    SUCCESS_MESSAGE,
    SAVE_ERROR_MESSAGE,
  },
  props: {
    maxNumberOfRepositoryDownloads: {
      type: Number,
      required: false,
      default: 0,
    },
    maxNumberOfRepositoryDownloadsWithinTimePeriod: {
      type: Number,
      required: false,
      default: 0,
    },
    gitRateLimitUsersAllowlist: {
      type: Array,
      required: false,
      default: () => [],
    },
  },
  data() {
    return {
      excludedUsers: this.gitRateLimitUsersAllowlist,
      numberOfRepos: this.maxNumberOfRepositoryDownloads,
      reportingTimePeriod: this.maxNumberOfRepositoryDownloadsWithinTimePeriod,
      formErrors: {
        numberOfRepos: '',
        reportingTimePeriod: '',
        excludedUsers: '',
      },
      isLoading: false,
    };
  },
  computed: {
    formHasError() {
      return Object.values(this.formErrors).some(Boolean);
    },
  },
  methods: {
    async updateGitAbuseRateLimitSettings() {
      this.isLoading = true;

      try {
        await Api.updateApplicationSettings({
          max_number_of_repository_downloads: this.numberOfRepos,
          max_number_of_repository_downloads_within_time_period: this.reportingTimePeriod,
          git_rate_limit_users_allowlist: this.excludedUsers,
        });

        createFlash({
          message: this.$options.i18n.SUCCESS_MESSAGE,
          type: 'notice',
        });
      } catch (error) {
        createFlash({
          message: this.$options.i18n.SAVE_ERROR_MESSAGE,
          captureError: true,
          error,
        });
      } finally {
        this.isLoading = false;
      }
    },
    addToExcludedUsers(username) {
      this.excludedUsers.push(username);
      this.formErrors.excludedUsers = validateExcludedUsers(this.excludedUsers);
    },
    removeFromExcludedUsers(username) {
      this.excludedUsers = this.excludedUsers.filter((x) => x !== username);
      this.formErrors.excludedUsers = validateExcludedUsers(this.excludedUsers);
    },
    checkNumberOfRepos() {
      this.formErrors.numberOfRepos = validateNumberOfRepos(this.numberOfRepos);
    },
    checkReportingTimePeriod() {
      this.formErrors.reportingTimePeriod = validateReportingTimePeriod(this.reportingTimePeriod);
    },
  },
};
</script>
<template>
  <section>
    <gl-form @submit.prevent="updateGitAbuseRateLimitSettings">
      <gl-form-group
        :label="$options.i18n.NUM_REPO_LABEL"
        :description="$options.i18n.NUM_REPO_DESCRIPTION"
        label-for="number-of-repos"
        :state="!Boolean(formErrors.numberOfRepos)"
        :invalid-feedback="formErrors.numberOfRepos"
        data-testid="number-of-repos-group"
      >
        <gl-form-input
          id="number-of-repos"
          v-model="numberOfRepos"
          type="number"
          :class="{ 'is-invalid': Boolean(formErrors.numberOfRepos) }"
          data-testid="number-of-repos-input"
          @blur="checkNumberOfRepos"
        />
      </gl-form-group>
      <gl-form-group
        :label="$options.i18n.REPORTING_TIME_PERIOD_LABEL"
        label-for="reporting-time-period"
        :state="!Boolean(formErrors.reportingTimePeriod)"
        :invalid-feedback="formErrors.reportingTimePeriod"
        data-testid="reporting-time-period-group"
      >
        <gl-form-input
          id="reporting-time-period"
          v-model="reportingTimePeriod"
          type="number"
          :class="{ 'is-invalid': Boolean(formErrors.reportingTimePeriod) }"
          data-testid="reporting-time-period-input"
          @blur="checkReportingTimePeriod"
        />
      </gl-form-group>
      <gl-form-group
        :label="$options.i18n.EXCLUDED_USERS_LABEL"
        label-for="excluded-users"
        :state="!Boolean(formErrors.excludedUsers)"
        :invalid-feedback="formErrors.excludedUsers"
        data-testid="excluded-users-group"
      >
        <template #description>
          <div class="gl-mt-3">
            {{ $options.i18n.EXCLUDED_USERS_DESCRIPTION }}
          </div>
        </template>
        <users-allowlist
          :excluded-usernames="excludedUsers"
          @user-added="addToExcludedUsers"
          @user-removed="removeFromExcludedUsers"
        />
      </gl-form-group>
      <gl-button
        :loading="isLoading"
        type="submit"
        variant="confirm"
        category="primary"
        :disabled="formHasError"
      >
        {{ $options.i18n.SAVE_CHANGES }}
      </gl-button>
    </gl-form>
  </section>
</template>
