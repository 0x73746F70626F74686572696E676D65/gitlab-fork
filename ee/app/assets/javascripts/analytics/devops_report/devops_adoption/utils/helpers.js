/**
 * A helper function which accepts the enabledNamespaces,
 *
 * @param {Object} params the enabledNamespaces data, and the check for open modals
 *
 * @return {Boolean} a boolean to determine if table data should be polled
 */
export const shouldPollTableData = ({ enabledNamespaces, openModal }) => {
  if (openModal) {
    return false;
  } else if (!enabledNamespaces.length) {
    return true;
  }

  const anyPendingEnabledNamespaces = enabledNamespaces.some(
    (node) => node.latestSnapshot === null,
  );

  return anyPendingEnabledNamespaces;
};
