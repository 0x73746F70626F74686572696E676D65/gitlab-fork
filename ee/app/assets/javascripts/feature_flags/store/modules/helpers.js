// eslint-disable-next-line import/prefer-default-export
export const parseFeatureFlagsParams = params => ({
  operations_feature_flags: {
    name: params.name,
    description: params.description,
    active: true,
    scopes_attributes: params.scopes.map(scope => ({
      environment_scope: scope.name,
      active: scope.active,
    })),
  },
});
