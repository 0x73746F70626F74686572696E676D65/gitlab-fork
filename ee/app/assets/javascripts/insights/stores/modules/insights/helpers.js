import { CHART_TYPES } from 'ee/insights/constants';
import { __ } from '~/locale';

const getGroupByValue = (queryParams) => {
  return queryParams.group_by || queryParams.params.group_by;
};

const getTypeValue = (queryParams) => {
  return queryParams.issuable_type || queryParams.params.issuable_type;
};

const getAxisTitle = (label) => {
  switch (label) {
    case 'day':
      return __('Days');
    case 'week':
      return __('Weeks');
    case 'month':
      return __('Months');
    case 'issue':
      return __('Issues');
    case 'merge_request':
      return __('Merge requests');
    default:
      return '';
  }
};

export const transformChartDataForGlCharts = (
  { type, query: queryParams },
  { labels, datasets },
) => {
  const formattedData = {
    xAxisTitle: getAxisTitle(getGroupByValue(queryParams)),
    yAxisTitle: getAxisTitle(getTypeValue(queryParams)),
    labels,
    datasets: [],
    seriesNames: [],
  };

  switch (type) {
    case CHART_TYPES.BAR:
      formattedData.datasets = [
        {
          name: 'all',
          data: labels.map((label, i) => [label, datasets[0].data[i]]),
        },
      ];
      break;
    case CHART_TYPES.STACKED_BAR:
      formattedData.datasets.push(
        ...datasets.map((dataset) => ({
          name: dataset.label,
          data: dataset.data,
        })),
      );
      break;
    case CHART_TYPES.LINE:
      formattedData.datasets.push(
        ...datasets.map((dataset) => ({
          name: dataset.label,
          data: labels.map((label, i) => [label, dataset.data[i]]),
        })),
      );

      break;
    default:
      formattedData.datasets = { all: labels.map((label, i) => [label, datasets[0].data[i]]) };
  }
  return formattedData;
};

export default {
  transformChartDataForGlCharts,
};
