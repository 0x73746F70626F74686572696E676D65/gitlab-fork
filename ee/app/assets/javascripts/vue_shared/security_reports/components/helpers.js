import { EMPTY_BODY_MESSAGE } from './constants';

/**
 * A helper function which validates the passed
 * in body string.
 *
 * It returns an empty string if the body has explicitly
 * been passed in as an empty string, a fallback
 * message if the body is null / undefined, else
 * it will return the original body string.
 *
 * @param {String} body the body message
 *
 * @return {String} the validated body message
 */
export const bodyWithFallBack = (body) => (body === '' ? '' : body || EMPTY_BODY_MESSAGE);

export const getCreatedIssueForVulnerability = (vulnerability) =>
  vulnerability.issue_links?.find((link) => link.link_type === 'created');

export const getDismissalTransitionForVulnerability = (vulnerability) => {
  const latestTransition = vulnerability.state_transitions?.at(-1);
  return latestTransition?.to_state.toLowerCase() === 'dismissed' ? latestTransition : null;
};
