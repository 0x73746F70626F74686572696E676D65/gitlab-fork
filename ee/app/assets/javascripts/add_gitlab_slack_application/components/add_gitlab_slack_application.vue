<script>
/* eslint-disable @gitlab/vue-require-i18n-strings */
import { GlButton, GlIcon } from '@gitlab/ui';
import createFlash from '~/flash';
import { redirectTo } from '~/lib/utils/url_utility';
import { __ } from '~/locale';

import GitlabSlackService from '../services/gitlab_slack_service';

export default {
  components: {
    GlButton,
    GlIcon,
  },

  props: {
    projects: {
      type: Array,
      required: false,
      default: () => [],
    },

    isSignedIn: {
      type: Boolean,
      required: true,
    },

    gitlabForSlackGifPath: {
      type: String,
      required: true,
    },

    signInPath: {
      type: String,
      required: true,
    },

    slackLinkPath: {
      type: String,
      required: true,
    },

    gitlabLogoPath: {
      type: String,
      required: true,
    },

    slackLogoPath: {
      type: String,
      required: true,
    },

    docsPath: {
      type: String,
      required: true,
    },
  },

  data() {
    return {
      selectedProjectId: this.projects && this.projects.length ? this.projects[0].id : 0,
    };
  },

  computed: {
    hasProjects() {
      return this.projects.length > 0;
    },
  },

  methods: {
    addToSlack() {
      GitlabSlackService.addToSlack(this.slackLinkPath, this.selectedProjectId)
        .then((response) => redirectTo(response.data.add_to_slack_link))
        .catch(() =>
          createFlash({
            message: __('Unable to build Slack link.'),
          }),
        );
    },
  },
};
</script>

<template>
  <div class="gitlab-slack-body gl-mx-auto gl-mt-11 gl-text-center">
    <div v-once class="gl-my-5 gl-display-flex gl-justify-content-center gl-align-items-center">
      <img :src="gitlabLogoPath" class="gl-h-11 gl-w-11" />
      <gl-icon name="arrow-right" :size="32" class="gl-mx-5 gl-text-gray-200" />
      <img :src="slackLogoPath" class="gitlab-slack-slack-logo gl-h-11 gl-w-11" />
    </div>

    <h1>{{ s__('SlackIntegration|GitLab for Slack') }}</h1>
    <p>{{ s__('SlackIntegration|Select a GitLab project to link with your Slack workspace.') }}</p>

    <div class="mx-auto prepend-top-20 text-center">
      <div v-if="isSignedIn && hasProjects">
        <select v-model="selectedProjectId" class="js-project-select form-control gl-mt-3 gl-mb-3">
          <option v-for="project in projects" :key="project.id" :value="project.id">
            {{ project.name }}
          </option>
        </select>

        <div class="gl-display-flex gl-justify-content-end">
          <gl-button category="primary" variant="confirm" class="float-right" @click="addToSlack">
            {{ __('Continue') }}
          </gl-button>
        </div>
      </div>

      <span v-else-if="isSignedIn && !hasProjects" class="js-no-projects">{{
        __("You don't have any projects available.")
      }}</span>

      <span v-else>
        You have to
        <a v-once :href="signInPath" class="js-gitlab-slack-sign-in-link">{{ __('log in') }}</a>
      </span>
    </div>

    <div class="center prepend-top-20 gl-mb-3 gl-mr-2 gl-ml-2">
      <img v-once :src="gitlabForSlackGifPath" class="gl-w-full" />
    </div>

    <div v-once class="text-center">
      <h3>{{ __('How it works') }}</h3>

      <div class="well gitlab-slack-well mx-auto">
        <code class="code mx-auto gl-mb-3"
          >/gitlab &lt;project-alias&gt; issue show &lt;id&gt;</code
        >
        <span>
          <gl-icon name="arrow-right" class="gl-mr-2 gl-text-gray-200" />
          Shows the issue with id <strong>&lt;id&gt;</strong>
        </span>

        <div class="gl-mt-3">
          <a v-once :href="docsPath">{{ __('More Slack commands') }}</a>
        </div>
      </div>
    </div>
  </div>
</template>
