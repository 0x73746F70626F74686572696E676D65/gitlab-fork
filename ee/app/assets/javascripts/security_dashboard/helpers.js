import isPlainObject from 'lodash/isPlainObject';
import { s__ } from '~/locale';
import { convertObjectPropsToSnakeCase } from '~/lib/utils/common_utils';
import { ALL, BASE_FILTERS } from 'ee/security_dashboard/store/modules/filters/constants';
import { REPORT_TYPES, SEVERITY_LEVELS } from 'ee/security_dashboard/store/constants';
import { VULNERABILITY_STATES } from 'ee/vulnerabilities/constants';

const parseOptions = obj =>
  Object.entries(obj).map(([id, name]) => ({ id: id.toUpperCase(), name }));

export const mapProjects = projects =>
  projects.map(p => ({ id: p.id.split('/').pop(), name: p.name }));

export const initFirstClassVulnerabilityFilters = projects => {
  const filters = [
    {
      name: s__('SecurityReports|Status'),
      id: 'state',
      options: [
        { id: ALL, name: s__('VulnerabilityStatusTypes|All') },
        ...parseOptions(VULNERABILITY_STATES),
      ],
      selection: new Set([ALL]),
    },
    {
      name: s__('SecurityReports|Severity'),
      id: 'severity',
      options: [BASE_FILTERS.severity, ...parseOptions(SEVERITY_LEVELS)],
      selection: new Set([ALL]),
    },
    {
      name: s__('SecurityReports|Report type'),
      id: 'reportType',
      options: [BASE_FILTERS.report_type, ...parseOptions(REPORT_TYPES)],
      selection: new Set([ALL]),
    },
  ];

  if (Array.isArray(projects)) {
    filters.push({
      name: s__('SecurityReports|Project'),
      id: 'projectId',
      options: [BASE_FILTERS.project_id, ...mapProjects(projects)],
      selection: new Set([ALL]),
    });
  }

  return filters;
};

export const getFormattedSummary = (rawSummary = {}) => {
  if (!isPlainObject(rawSummary)) {
    return [];
  }
  // Convert keys to snake case so they can be matched against REPORT_TYPES keys for translation
  const snakeCasedSummary = convertObjectPropsToSnakeCase(rawSummary);
  // Convert object to an array of entries to make it easier to loop through
  const summaryEntries = Object.entries(snakeCasedSummary);
  // Filter out empty entries as we don't want to display those in the summary
  const withoutEmptyEntries = summaryEntries.filter(
    ([, scanSummary]) => scanSummary?.vulnerabilitiesCount !== undefined,
  );
  // Replace keys with translations found in REPORT_TYPES if available
  const formattedEntries = withoutEmptyEntries.map(([scanType, scanSummary]) => {
    const name = REPORT_TYPES[scanType];
    return name ? [name, scanSummary] : null;
  });
  // Filter out keys that could not be matched with any translation and are thus considered invalid
  return formattedEntries.filter(entry => entry !== null);
};

export default () => ({});
