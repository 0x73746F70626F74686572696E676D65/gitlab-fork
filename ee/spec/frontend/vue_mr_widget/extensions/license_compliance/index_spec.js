import MockAdapter from 'axios-mock-adapter';
import { mountExtended } from 'helpers/vue_test_utils_helper';
import waitForPromises from 'helpers/wait_for_promises';
import axios from '~/lib/utils/axios_utils';
import extensionsContainer from '~/vue_merge_request_widget/components/extensions/container';
import { registerExtension } from '~/vue_merge_request_widget/components/extensions';
import licenseComplianceExtension from 'ee/vue_merge_request_widget/extensions/license_compliance';
import httpStatusCodes from '~/lib/utils/http_status';
import {
  licenseComplianceSuccess,
  licenseComplianceRemovedLicenses,
  licenseComplianceNewAndRemovedLicenses,
  licenseComplianceEmpty,
} from './mock_data';

describe('License Compliance extension', () => {
  let wrapper;
  let mock;

  registerExtension(licenseComplianceExtension);

  const endpoint = '/group-name/project-name/-/merge_requests/78/license_scanning_reports';

  const mockApi = (statusCode, data) => {
    mock.onGet(endpoint).reply(statusCode, data);
  };

  const findToggleCollapsedButton = () => wrapper.findByTestId('toggle-button');
  const findAllExtensionListItems = () => wrapper.findAllByTestId('extension-list-item');

  const createComponent = () => {
    wrapper = mountExtended(extensionsContainer, {
      propsData: {
        mr: {
          licenseCompliance: {
            license_scanning_comparison_path: endpoint,
            api_approvals_path: endpoint,
          },
        },
      },
    });
  };

  beforeEach(() => {
    mock = new MockAdapter(axios);
  });

  afterEach(() => {
    wrapper.destroy();
    mock.restore();
  });

  describe('summary', () => {
    it('displays loading text', () => {
      mockApi(httpStatusCodes.OK, licenseComplianceSuccess);

      createComponent();

      expect(wrapper.text()).toBe('License Compliance test metrics results are being parsed');
    });

    it('displays failed loading text', async () => {
      mockApi(httpStatusCodes.INTERNAL_SERVER_ERROR);

      createComponent();

      await waitForPromises();
      expect(wrapper.text()).toBe('License Compliance failed loading results');
    });

    it('displays no licenses', async () => {
      mockApi(httpStatusCodes.OK, licenseComplianceEmpty);

      createComponent();

      await waitForPromises();

      expect(wrapper.text()).toBe(
        'License Compliance detected no licenses for the source branch only',
      );
    });

    it('displays new licenses count', async () => {
      mockApi(httpStatusCodes.OK, licenseComplianceSuccess);

      createComponent();

      await waitForPromises();

      expect(wrapper.text()).toBe(
        'License Compliance detected 3 licenses for the source branch only',
      );
    });

    it('displays removed licenses count', async () => {
      mockApi(httpStatusCodes.OK, licenseComplianceRemovedLicenses);

      createComponent();

      await waitForPromises();

      expect(wrapper.text()).toBe(
        'License Compliance detected no licenses for the source branch only',
      );
    });

    it('displays new and removed licenses count', async () => {
      mockApi(httpStatusCodes.OK, licenseComplianceNewAndRemovedLicenses);

      createComponent();

      await waitForPromises();

      expect(wrapper.text()).toBe(
        'License Compliance detected 3 licenses for the source branch only',
      );
    });
  });

  describe('expanded data', () => {
    describe('with new licesnes', () => {
      beforeEach(async () => {
        mockApi(httpStatusCodes.OK, licenseComplianceSuccess);

        createComponent();

        await waitForPromises();

        findToggleCollapsedButton().trigger('click');

        await waitForPromises();
      });

      it('displays denied licenses', async () => {
        expect(findAllExtensionListItems().at(0).element).toMatchSnapshot();
      });

      it('displays uncategorized licenses', async () => {
        expect(findAllExtensionListItems().at(1).element).toMatchSnapshot();
      });

      it('displays allowed licenses', async () => {
        expect(findAllExtensionListItems().at(2).element).toMatchSnapshot();
      });
    });
  });
});
