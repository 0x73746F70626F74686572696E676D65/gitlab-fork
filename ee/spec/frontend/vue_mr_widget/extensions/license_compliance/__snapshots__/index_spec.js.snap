// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`License Compliance extension expanded data with new licenses displays allowed licenses 1`] = `
<div
  class="gl-py-3 gl-pl-7"
  data-testid="extension-list-item"
>
  <div
    class="gl-w-full"
  >
    <div
      class="gl-w-full"
    >
      <div
        class="gl-mb-2"
      >
        <strong>
          Allowed
        </strong>
      </div>
       
      <div
        class="gl-display-flex"
      >
        <!---->
         
        <div
          class="gl-w-full"
        >
          <div
            class="gl-display-flex gl-flex-nowrap"
          >
            <div
              class="gl-flex-wrap gl-display-flex gl-w-full"
            >
              <div
                class="gl-display-flex gl-align-items-center"
              >
                <p
                  class="gl-m-0"
                >
                  Acceptable for use in this project
                </p>
              </div>
               
              <!---->
               
              <!---->
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="gl-ml-auto gl-pl-3"
            >
              <!---->
               
              <!---->
            </div>
          </div>
           
          <!---->
        </div>
      </div>
       
      <ul
        class="gl-m-0 gl-p-0 gl-list-style-none"
      >
        <li>
          <div
            class="gl-w-full gl-pl-6"
            data-testid="child-content"
          >
            <!---->
             
            <div
              class="gl-display-flex"
            >
              <div
                class="gl-rounded-full gl-mr-3 gl-relative gl-p-2 gl-pl-0 gl-text-green-500"
              >
                <svg
                  aria-label="Success "
                  class="gl-display-block gl-icon s12"
                  data-testid="status-success-icon"
                  role="img"
                >
                  <use
                    href="#status-success"
                  />
                </svg>
              </div>
               
              <div
                class="gl-w-full"
              >
                <div
                  class="gl-display-flex gl-flex-nowrap"
                >
                  <div
                    class="gl-flex-wrap gl-display-flex gl-w-full"
                  >
                    <div
                      class="gl-display-flex gl-align-items-center"
                    >
                      <p
                        class="gl-m-0"
                      />
                    </div>
                     
                    <div
                      class="gl-pr-2"
                    >
                      <a
                        class="gl-link"
                        href="https://opensource.org/licenses/ISC"
                      >
                        ISC License
                      </a>
                    </div>
                     
                    <!---->
                     
                    <div>
                      <p
                        class="gl-m-0"
                      >
                         
                        Used by abbrev, anymatch
                      </p>
                    </div>
                     
                    <!---->
                  </div>
                   
                  <div
                    class="gl-ml-auto gl-pl-3"
                  >
                    <!---->
                     
                    <!---->
                  </div>
                </div>
                 
                <!---->
              </div>
            </div>
             
            <!---->
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`License Compliance extension expanded data with new licenses displays denied licenses 1`] = `
<div
  class="gl-py-3 gl-pl-7 gl-border-b-solid gl-border-b-1 gl-border-gray-100"
  data-testid="extension-list-item"
>
  <div
    class="gl-w-full"
  >
    <div
      class="gl-w-full"
    >
      <div
        class="gl-mb-2"
      >
        <strong>
          Denied
        </strong>
      </div>
       
      <div
        class="gl-display-flex"
      >
        <!---->
         
        <div
          class="gl-w-full"
        >
          <div
            class="gl-display-flex gl-flex-nowrap"
          >
            <div
              class="gl-flex-wrap gl-display-flex gl-w-full"
            >
              <div
                class="gl-display-flex gl-align-items-center"
              >
                <p
                  class="gl-m-0"
                >
                  Out-of-compliance with the project's policies and should be removed
                </p>
              </div>
               
              <!---->
               
              <!---->
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="gl-ml-auto gl-pl-3"
            >
              <!---->
               
              <!---->
            </div>
          </div>
           
          <!---->
        </div>
      </div>
       
      <ul
        class="gl-m-0 gl-p-0 gl-list-style-none"
      >
        <li>
          <div
            class="gl-w-full gl-pl-6"
            data-testid="child-content"
          >
            <!---->
             
            <div
              class="gl-display-flex"
            >
              <div
                class="gl-rounded-full gl-mr-3 gl-relative gl-p-2 gl-pl-0 gl-text-red-500"
              >
                <svg
                  aria-label="Failed "
                  class="gl-display-block gl-icon s12"
                  data-testid="status-failed-icon"
                  role="img"
                >
                  <use
                    href="#status-failed"
                  />
                </svg>
              </div>
               
              <div
                class="gl-w-full"
              >
                <div
                  class="gl-display-flex gl-flex-nowrap"
                >
                  <div
                    class="gl-flex-wrap gl-display-flex gl-w-full"
                  >
                    <div
                      class="gl-display-flex gl-align-items-center"
                    >
                      <p
                        class="gl-m-0"
                      />
                    </div>
                     
                    <div
                      class="gl-pr-2"
                    >
                      <a
                        class="gl-link"
                        href="https://opensource.org/licenses/Apache-2.0"
                      >
                        Apache License 2.0
                      </a>
                    </div>
                     
                    <!---->
                     
                    <div>
                      <p
                        class="gl-m-0"
                      >
                         
                        Used by websocket-driver, websocket-extensions, xml-name-validator
                      </p>
                    </div>
                     
                    <!---->
                  </div>
                   
                  <div
                    class="gl-ml-auto gl-pl-3"
                  >
                    <!---->
                     
                    <!---->
                  </div>
                </div>
                 
                <!---->
              </div>
            </div>
             
            <!---->
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`License Compliance extension expanded data with new licenses displays uncategorized licenses 1`] = `
<div
  class="gl-py-3 gl-pl-7 gl-border-b-solid gl-border-b-1 gl-border-gray-100"
  data-testid="extension-list-item"
>
  <div
    class="gl-w-full"
  >
    <div
      class="gl-w-full"
    >
      <div
        class="gl-mb-2"
      >
        <strong>
          Uncategorized
        </strong>
      </div>
       
      <div
        class="gl-display-flex"
      >
        <!---->
         
        <div
          class="gl-w-full"
        >
          <div
            class="gl-display-flex gl-flex-nowrap"
          >
            <div
              class="gl-flex-wrap gl-display-flex gl-w-full"
            >
              <div
                class="gl-display-flex gl-align-items-center"
              >
                <p
                  class="gl-m-0"
                >
                  No policy matches this license
                </p>
              </div>
               
              <!---->
               
              <!---->
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="gl-ml-auto gl-pl-3"
            >
              <!---->
               
              <!---->
            </div>
          </div>
           
          <!---->
        </div>
      </div>
       
      <ul
        class="gl-m-0 gl-p-0 gl-list-style-none"
      >
        <li>
          <div
            class="gl-w-full gl-pl-6"
            data-testid="child-content"
          >
            <!---->
             
            <div
              class="gl-display-flex"
            >
              <div
                class="gl-rounded-full gl-mr-3 gl-relative gl-p-2 gl-pl-0 gl-text-gray-500"
              >
                <svg
                  aria-label="Notice "
                  class="gl-display-block gl-icon s12"
                  data-testid="status-alert-icon"
                  role="img"
                >
                  <use
                    href="#status-alert"
                  />
                </svg>
              </div>
               
              <div
                class="gl-w-full"
              >
                <div
                  class="gl-display-flex gl-flex-nowrap"
                >
                  <div
                    class="gl-flex-wrap gl-display-flex gl-w-full"
                  >
                    <div
                      class="gl-display-flex gl-align-items-center"
                    >
                      <p
                        class="gl-m-0"
                      />
                    </div>
                     
                    <div
                      class="gl-pr-2"
                    >
                      <a
                        class="gl-link"
                        href="http://opensource.linux-mirror.org/licenses/afl-2.1.txt"
                      >
                        Academic Free License v2.1
                      </a>
                    </div>
                     
                    <!---->
                     
                    <div>
                      <p
                        class="gl-m-0"
                      >
                         
                        Used by json-schema
                      </p>
                    </div>
                     
                    <!---->
                  </div>
                   
                  <div
                    class="gl-ml-auto gl-pl-3"
                  >
                    <!---->
                     
                    <!---->
                  </div>
                </div>
                 
                <!---->
              </div>
            </div>
             
            <!---->
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
